python_modules/dagster/dagster/_core/definitions/decorators/asset_check_decorator.py

```diff
@@ -0,0 +1,151 @@
+            if it does not. If not set, Dagster will accept any config provided for the op.
+        op_tags (Optional[Dict[str, Any]]): A dictionary of tags for the op that executes the check.
+            Frameworks may expect and require certain metadata to be attached to a op. Values that
+            are not strings will be json encoded and must meet the criteria that
+            `json.loads(json.dumps(value)) == value`.
+        compute_kind (Optional[str]): A string to represent the kind of computation that executes
+            the check, e.g. "dbt" or "spark".
+        retry_policy (Optional[RetryPolicy]): The retry policy for the op that executes the check.
+
+    Produces an :py:class:`AssetChecksDefinition` object.
+
+    Examples:
+        .. code-block:: python
+            from dagster import asset, asset_check, AssetCheckResult
+
+            @asset
+            def my_asset() -> None:
+                ...
+
+            @asset_check(asset=my_asset, description="Check that my asset has enough rows")
+            def my_asset_has_enough_rows() -> AssetCheckResult:
+                num_rows = ...
+                return AssetCheckResult(success=num_rows > 5, metadata={"num_rows": num_rows})
+
+
+        .. code-block:: python
+            from dagster import asset, asset_check, AssetCheckResult
+            from pandas import DataFrame
+
+            @asset
+            def my_asset() -> DataFrame:
+                ...
+
+            @asset_check(description="Check that my asset has enough rows")
+            def my_asset_has_enough_rows(my_asset: DataFrame) -> AssetCheckResult:
+                num_rows = my_asset.shape[0]
+                return AssetCheckResult(success=num_rows > 5, metadata={"num_rows": num_rows})
+    """
+
+    def inner(fn: AssetCheckFunction) -> AssetChecksDefinition:
+        check.callable_param(fn, "fn")
+        resolved_name = name or fn.__name__
+        asset_key = asset_key_from_coercible_or_definition(asset) if asset is not None else None
+
+        out = Out(dagster_type=None)
+        input_tuples_by_asset_key = build_asset_ins(
+            fn, {}, {asset_key} if asset_key is not None else set()
+        )
+        if len(input_tuples_by_asset_key) == 0:
+            raise DagsterInvalidDefinitionError(
+                f"No target asset provided when defining check '{resolved_name}'"
+            )
+
+        if len(input_tuples_by_asset_key) > 1:
+            raise DagsterInvalidDefinitionError(
+                f"When defining check '{resolved_name}', Multiple target assets provided:"
+                f" {[key.to_user_string() for key in input_tuples_by_asset_key.keys()]}. Only one"
+                " is allowed."
+            )
+
+        resolved_asset_key = next(iter(input_tuples_by_asset_key.keys()))
+        op_def = _Op(
+            name=f"{resolved_asset_key.to_python_identifier()}_{resolved_name}",
```

Comment:
might want to consider adding a "check" or "__check" suffix to avoid any potential ambiguity with asset keys that look similar

Reply:
how strongly do you feel on this one? do you have an ambiguous example in mind? a little worried about adding clutter to the places where we list steps, like the run view, but could easily be swayed

Reply:
Not strongly at all

Reply:
Also easy to change

Reply:
👍